provider/webhook/api/httpapi.go

```diff
@@ -85,18 +100,20 @@ func (p *WebhookServer) AdjustEndpointsHandler(w http.ResponseWriter, req *http.
 		return
 	}
 
-	var pve []*endpoint.Endpoint
+	var pve []*apiv1alpha1.Endpoint
 	if err := json.NewDecoder(req.Body).Decode(&pve); err != nil {
 		log.Errorf("Failed to decode in adjustEndpointsHandler: %v", err)
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 	w.Header().Set(ContentTypeHeader, MediaTypeFormatAndVersion)
-	pve, err := p.Provider.AdjustEndpoints(pve)
+	endpoints := adapter.ToInternalEndpoints(pve)
+	endpoints, err := p.Provider.AdjustEndpoints(endpoints)
 	if err != nil {
 		log.Errorf("Failed to call adjust endpoints: %v", err)
 		w.WriteHeader(http.StatusInternalServerError)
 	}
+	pve = adapter.ToAPIEndpoints(endpoints)
```

Comment:
It was not covered before, which probably explains why tests were green when there was an issue with webhook

<img width="969" height="379" alt="Screenshot 2025-09-25 at 10 34 41" src="https://github.com/user-attachments/assets/1e73d3b6-1b7e-4a2b-9c58-b69f13a87fe9" />